### PR TITLE
Update Description, Project and Tag in one func

### PR DESCRIPTION
**Pull request by NghiaTranUIT**
### 📒 Description
This PR improves how we update the TE from TimeEntryAutoComplete by reducing some unnecessary UI Update from the library from 4 -> 2 times:
- 1 call is for updating the UI from AutoComplete Item
- 1 call is from the Syncer

### 🕶️ Types of changes
 **New feature** (non-breaking change which adds functionality) 

### 🤯 List of changes
- [x] Introduce update_ui in save func to only update the UI if need
- [x] Add 1 func to update TE from autocomplete item (Description, project, task, tags)
- [x] Use this func to update on mac build

### 👫 Relationships
- macOS part of [Library] Update Description, Project and Tag in one func #3244

#### All Features are remains
- Play around and make sure we're able to update the TE as usual

#### There are only 2 UI Updates from Library
1. Open Editor, and search some TE in Description Text Field
2. Select any TE AutoComplete Item in Description
3. Observer that there are only 2 UI Update from the library by looking at the Console
```
❌ TimeEntryListViewController displayTimeEntryList
```

### CODE REVIEW

**Comment by NghiaTranUIT** [comment]
That's total make sense @tanel 👍 Thank for the keyword "Flag argument"

I will address it 💻

**Comment by NghiaTranUIT** [comment]
Ready for reviewing @skel35 👍

### How it's addressed
> This is a FlagArgument anti-pattern https://martinfowler.com/bliki/FlagArgument.html. I know we have a few of those in our codebase, but I think it's bad and we shouldn't introduce more.

I completely remove the introduction of `bool update` param in `save()` function

> The logic is moved into the library, but we still perform the same actions (like finding the entry by GUID) multiple times. If we perform them once it will improve the performance.

- Split and reuse the logic when updating description, project, tag, and billable
- Introduce big function to update all things with one single update and one time to find TimeEntry from GUID
```c++
error Context::updateTimeEntry(
                       const std::string &GUID,
                       const std::string &description,
                       const Poco::UInt64 task_id,
                       const Poco::UInt64 project_id,
                       const std::string &project_guid,
                       const std::string &tags,
                     const bool billable) 
```

> (minor) I believe this is a bug that the macOS app does not update the billable flag, so I think updating it should be included in this method as well.

Update billable param in the same function

File: src/ui/osx/TogglDesktop/Feature/TimeEntryEditor/DesktopLibraryBridge.m [hunk 1]
```diff
@@ -192,15 +192,15 @@ - (void) registerUndoDeleteItem:(TimeEntryViewItem *)item
...
-	[self updateTimeEntryWithTags:autocomplete.tags guid:timeEntry.GUID];
+    const char *tags = [[autocomplete.tags componentsJoinedByString:@"\t"] UTF8String];
+    toggl_set_time_entry(ctx,
+                         [timeEntry.GUID UTF8String],
+                         [autocomplete.Description UTF8String],
+                         autocomplete.TaskID,
+                         autocomplete.ProjectID,
+                         0,
+                         tags,
+                         timeEntry.billable);
```
Comment (skel35) [inline src/ui/osx/TogglDesktop/Feature/TimeEntryEditor/DesktopLibraryBridge.m]: `billable` should come from `autocomplete` and not from `timeEntry`.
We need this to make sure autocompleting billable time entries sets the billable flag correctly. I believe the fact that we didn't do it before was a bug on macOS.

**Comment by NghiaTranUIT** [comment]
Thank you for your feedback @skel35. It's done for reviewing 👍

File: src/context.h [hunk 2]
```diff
@@ -388,6 +388,15 @@ class TOGGL_INTERNAL_EXPORT Context : public TimelineDatasource {
         const Poco::UInt64 project_id,
         const std::string &project_guid);
 
+    error updateTimeEntry(
```
Comment (skel35) [inline src/context.h]: this is a public method so it should be capitalized

Comment (NghiaTranUIT) [reply src/context.h]: Thanks for pointing out 👍

**Pull request merged**